narwhals/series.py

```diff
@@ -99,6 +99,30 @@ def cast(
         )
 
     def mean(self) -> Any:
+        """
+        Reduce this Series to the mean value.
+
+        Examples:
+            >>> import pandas as pd
+            >>> import polars as pl
+            >>> import narwhals as nw
+            >>> s = [1, 2, 3]
+            >>> s_pd = pd.Series(s)
+            >>> s_pl = pl.Series(s)
+
+            We define a library agnostic function:
+
+            >>> def func(s_any):
+            ...     s = nw.from_native(s_any, series_only=True)
+            ...     return s.mean()
+
+            We can then pass pandas, Polars or any other library Series to `func`:
```

Comment:
'any other library' might be a bit too much 😄 shall we just say "any other Narwhals-compliant library"? or just leave it as "pandas or Polars" and we'll sort out the "any other library" part later in a large pull request

Reply:
Right... 😅

Reply:
Okay